bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/local/handlers/TuyaDecoder.java

```diff
@@ -170,18 +196,23 @@ public void decode(@NonNullByDefault({}) ChannelHandlerContext ctx, @NonNullByDe
             m = new MessageWrapper<>(commandType,
                     Objects.requireNonNull(gson.fromJson(new String(payload), DiscoveryMessage.class)));
         } else {
-            byte[] decodedMessage = protocol == V3_4 ? CryptoUtil.decryptAesEcb(payload, sessionKey, true)
-                    : CryptoUtil.decryptAesEcb(payload, sessionKey, false);
+            byte[] decodedMessage = protocol == V3_5 ? CryptoUtil.decryptAesGcm(payload, sessionKey, header, null) : protocol == V3_4 ? CryptoUtil.decryptAesEcb(payload, sessionKey, true) : CryptoUtil.decryptAesEcb(payload, sessionKey, false);
```

Comment:
That's really hard to read. What do you think of using an enhanced switch statement?

```
byte[] decodedMessage = switch (protocol) {
    case V3_5 -> CryptoUtil.decryptAesGcm(payload, sessionKey, header, null);
    case V3_4 -> CryptoUtil.decryptAesEcb(payload, sessionKey, true);
    default -> CryptoUtil.decryptAesEcb(payload, sessionKey, false);
}
```